tests/test_cnn.py

```diff
@@ -35,3 +38,113 @@ def test_cnn(tmp_path, model_class):
     assert np.allclose(action, model.predict(obs, deterministic=True)[0])
 
     os.remove(str(tmp_path / SAVE_NAME))
+
+
+def patch_dqn_names_(model):
+    # Small hack to make the test work with DQN
+    if isinstance(model, DQN):
+        model.critic = model.q_net
+        model.critic_target = model.q_net_target
+
+
+def params_should_match(params, other_params):
+    for param, other_param in zip_strict(params, other_params):
+        assert th.allclose(param, other_param)
+
+
+def params_should_differ(params, other_params):
+    for param, other_param in zip_strict(params, other_params):
+        assert not th.allclose(param, other_param)
+
+
+def check_td3_feature_extractor_match(model):
+    for (key, actor_param), critic_param in zip(model.actor_target.named_parameters(), model.critic_target.parameters()):
+        if "features_extractor" in key:
+            assert th.allclose(actor_param, critic_param)
+
+
+@pytest.mark.parametrize("model_class", [SAC, TD3, DQN])
+def test_feature_extractor_target_net(model_class):
+    env = FakeImageEnv(screen_height=40, screen_width=40, n_channels=1, discrete=model_class not in {SAC, TD3})
+    # Avoid memory error when using replay buffer
+    # Reduce the size of the features
+    kwargs = dict(buffer_size=250, learning_starts=100, policy_kwargs=dict(features_extractor_kwargs=dict(features_dim=32)))
+    model = model_class("CnnPolicy", env, seed=0, **kwargs)
+
+    patch_dqn_names_(model)
+
+    # Critic and target should be equal at the begginning of training
+    params_should_match(model.critic.parameters(), model.critic_target.parameters())
+
+    # TD3 has also a target actor net
+    if model_class == TD3:
+        params_should_match(model.actor.parameters(), model.actor_target.parameters())
+
+    model.learn(200)
+
+    # Critic and target should differ
+    params_should_differ(model.critic.parameters(), model.critic_target.parameters())
+
+    if model_class == TD3:
+        params_should_differ(model.actor.parameters(), model.actor_target.parameters())
+
+        # Actor and critic feature extractor should be the same
+        check_td3_feature_extractor_match(model)
+
+    # Re-initialize and collect some random data (without doing gradient steps)
```

Comment:
```suggestion
    # Re-initialize and collect some random data (without doing gradient steps, since 10 < learning_starts = 100)
```
Was initially confused why there are no gradient steps here

Reply:
good point